crates/chain-config/src/config/coin.rs

```diff
@@ -33,6 +42,40 @@ pub struct CoinConfig {
     pub asset_id: AssetId,
 }
 
+impl NonSkippingSerialize for CoinConfig {
```

Comment:
Formats that don't include a schema cannot tolerate fields being skipped. Json can because from the encoded data it is clear that the field is missing. Postcard must read something for the optional field. Bincode as well. 

The problem is that the derived `Serialize` implementation skips over `Option<?>` fields if they are `None` so that we may have more friendly json encoding. But that causes the above described issue for `Postcard`. 

So you can either:
1. Don't skip the fields in the derived `Serialize`. This means that you need to manually encode the json if you want to retain the nice property of not having `null`s scattered about. 
2. Have the derived `Serialize` skip the fields but have another trait that, when implemented, gives a guarantee that the type either didn't skip a field or will provide a `Serialize` implementation that will not skip a field. 

I've tried option 1 and it turned out to be more verbose with all the .is_none checks.

Option 2 was easier to code. If a type's derived `Serialize` doesn't skip fields then we can just implement the thait `NonSkippingSerialize` that has a default implementation of the single method it declares.

If it does skip then it becomes a simple 'forward calls to .serialize for all fields of the struct'. No checks for null like option 1 warranted.

I've looked into the possibility of writing our own derive macro (Like `Bincode` did for this exact reason) but didn't know if this was enough of a painpoint to warrant a whole proc macro crate.